src/map/latent_effect_container.cpp

```diff
@@ -1674,7 +1675,23 @@ void CLatentEffectContainer::CheckLatentsZone()
             }
             break;
         case LATENT_WEATHER_ELEMENT:
-            if (zoneutils::GetWeatherElement(zoneutils::GetZone(m_POwner->getZone())->GetWeather()) == m_LatentEffectList.at(i)->GetConditionsValue())
+            // In the default case we need to pull weather from zone/player
+            if (weather == MAX_WEATHER_ID)
```

Comment:
`if (weather == MAX_WEATHER_ID)` is an impossible condition. If that ever happens, a debug break will be tripped.